src/chain.h

```diff
@@ -378,6 +378,7 @@ class CChain {
 
     /** Efficiently check whether a block is present in this chain. */
     bool Contains(const CBlockIndex *pindex) const {
+        if (!pindex) return false;  // null pointer isn't in this chain
```

Comment:
Is it expected to pass null to this function ? Considering the previous implementation, it doesn't looks like it is. Using DbgAssert seems more appropriate.